src/pgstac/sql/004_search.sql

```diff
@@ -760,11 +779,22 @@ DECLARE
     out_records jsonb;
     out_len int;
     _limit int := coalesce((_search->>'limit')::int, 10);
+    init_limit int := _limit;
     _querylimit int;
     _fields jsonb := coalesce(_search->'fields', '{}'::jsonb);
     has_prev boolean := FALSE;
     has_next boolean := FALSE;
+    items_cnt int := coalesce(jsonb_array_length(_search->'ids'),0);
 BEGIN
+    RAISE NOTICE 'Items Count: %', items_cnt;
+    IF items_cnt > 0 THEN
+        IF items_cnt <= _limit THEN
+            _limit := items_cnt - 1;
```

Comment:
Just following the logic here: the subtraction is so that when the search limit is bumped by 1 to check to see if it needs a "next token", it evens out and still doesn't keep scanning?

Reply:
yep, otherwise it would keep scanning for a next item which we know it wouldn't find